### ISSUE TITLE
[Bug]: "ERR Path '.' does not exist or not an array" error when adding a JSON field to a JSON with many fields

### ISSUE THREAD

**Issue by devtimlai** [RedisInsight/RedisInsight#3472]
**Preconditions** (Any important steps we need to know)
Redis Insight 2.50.0

**Steps to reproduce** (How to reproduce what you found step by step)
1. Add a JSON key with many fields. E.g. a JSON below:
```
{
  "A": false,
  "B": 200,
  "C": false,
  "D": 50,
  "E": 15,
  "F": 10,
  "G": 50,
  "H": 10,
  "I": 0.05,
  "J": 100,
  "K": 5,
  "L": 0.45,
  "M": 100,
  "O": 0.4,
  "P": {
    "A": {
      "A": 0.4,
      "B": 80000,
      "C": 2.5,
      "D": 100000
    },
    "B": {
      "A": 0.4,
      "B": 50000,
      "C": 2.5,
      "D": 55000
    },
    "C": {
      "A": 0.3,
      "B": 250,
      "C": 20,
      "D": -500
    },
    "D": {
      "A": 0.15,
      "B": 1000,
      "C": 0.5,
      "D": 5000
    },
    "E": {
      "A": 0.15,
      "B": 1000,
      "C": 0.5,
      "D": 5000
    },
    "F": {
      "A": 0.25,
      "B": 10,
      "C": null,
      "D": 35
    }
  },
  "Q": {
    "A": 1,
    "B": 1,
    "C": 1,
    "D": 1,
    "E": 1,
    "F": 1
  },
  "R": {
    "A": {
      "A": {
        "C 0": 0.01,
        "C 1": 0.01,
        "C S": 0.01,
        "BP 0": 0.01
      },
      "B": {
        "C 0": 0.01,
        "C 1": 0.01,
        "C S": 0.01,
        "BP 0": 0.01
      },
      "C": {
        "C 0": 0.01,
        "C 1": 0.01,
        "C S": 0.01,
        "BP 0": 0.01
      }
    }
  },
  "S": {
    "A": {
      "A": {
        "A": {}
      },
      "B": {
        "A": {}
      }
    }
  },
  "U": 10,
  "V": 10,
  "W": {
    "A": {
      "A": {
        "A": {
          "C 0": 1,
          "C 1": 1,
          "C 2": 1
        },
        "B": 1
      },
      "B": {
        "A": {
          "C 0": 1,
          "C 1": 1,
          "C 2": 1
        }
      }
    }
  }
}
```

2. Try to add a JSON field.
3. We can see that the JSON field is not added successfully. Error "ERR Path '.' does not exist or not an array" is shown.

If I try to add a JSON field on the JSON that does not have many fields, I don't see this "ERR Path '.' does not exist or not an array" error. For example:

1. Add a JSON with not many fields
```
{
  "A": 1,
  "B": 10,
  "C": "x"
}
```
2. Try to add a JSON field
3. Field is successfully added

**Actual behavior** (A short description of what you found)

Error "ERR Path '.' does not exist or not an array" is shown when adding a JSON field.

**Expected behavior** (A short description of what you expected to find)

Expect we could add the JSON field.

**Screenshots** (Paste or drag-and-drop a screenshot or a link to a recording)

![Screenshot 2024-06-08 at 2 19 42 AM](https://github.com/RedisInsight/RedisInsight/assets/36029194/eb93decb-7f3c-4496-839e-3f03b0513c39)

Screen recording for adding a JSON field to a JSON with **many** fields:

https://github.com/RedisInsight/RedisInsight/assets/36029194/6971d9d7-64aa-468d-a27a-de9a113877c5

Screen recording for adding a JSON field to a JSON with _not_ many fields:

https://github.com/RedisInsight/RedisInsight/assets/36029194/5a66daab-e512-4ae6-967f-3675075cc6f0

**Additional context** (Operating system, version of Redis Insight, Redis database version, Redis module version, database type, connection type, logs, or any other information)

This happens both on the Redis Insight in macOS and web.

**Comment by devtimlai** [comment]
@AaronForce1 FYI.

**Comment by devtimlai** [comment]
@ViktarStarastsenka 

```
> INFO
# Server
redis_version:7.2.4
```

**Comment by ViktarStarastsenka** [comment]
@devtimlai , can you please provide the full `Server` and `modules` sections?

**Comment by devtimlai** [comment]
@ViktarStarastsenka sure,

```
> INFO
# Server
redis_version:7.2.4
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:78acd54029a5a5bc
redis_mode:standalone
os:Linux 6.6.22-linuxkit aarch64
arch_bits:64
monotonic_clock:POSIX clock_gettime
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:11.4.0
process_id:9
process_supervised:no
run_id:78ce6715be24268617f9d1031d4c9c454a7f12a9
tcp_port:6379
server_time_usec:1718085124951406
uptime_in_seconds:447611
uptime_in_days:5
hz:10
configured_hz:10
lru_clock:6809092
executable:/opt/redis-stack/bin/redis-server
config_file:
io_threads_active:0
listener0:name=tcp,bind=*,bind=-::*,port=6379

# Modules
module:name=ReJSON,ver=20610,api=1,filters=0,usedby=[search],using=[],options=[handle-io-errors]
module:name=timeseries,ver=11012,api=1,filters=0,usedby=[],using=[],options=[]
module:name=search,ver=20813,api=1,filters=0,usedby=[],using=[ReJSON],options=[handle-io-errors]
module:name=redisgears_2,ver=20020,api=1,filters=0,usedby=[],using=[],options=[]
module:name=RedisCompat,ver=1,api=1,filters=0,usedby=[],using=[],options=[]
module:name=bf,ver=20612,api=1,filters=0,usedby=[],using=[],options=[]
```

**Comment by ViktarStarastsenka** [comment]
@devtimlai , this should be fixed with 2.52 (released today).
